File: my-life-backend/services/finance-service/src/main/java/com/andreas/mylife/finance/services/AccountService.java
package com.andreas.mylife.finance.services;

import java.util.List;
import java.util.UUID;

import com.andreas.mylife.finance.dto.request.AccountRequest;
import com.andreas.mylife.finance.dto.response.AccountResponse;

public interface AccountService {

    AccountResponse createAccount(UUID userId, AccountRequest request);

    List<AccountResponse> getAccountsByUserId(UUID userId);
}

File: my-life-app/services/finance/AccountService.ts
import { http } from "@/http/http_helper"
import { ApiResponse } from "@/types/api_response"
import { AccountRequest } from "@/types/dto/finance/request/account_request";
import { AccountResponse } from "@/types/dto/finance/response/account_response"

const BASE = "/api/proxy/finance-service/account"

export class AccountService {
    static getUserAccounts(){
        return http.get<ApiResponse<AccountResponse[]>> (`${BASE}`);
    }
    
    static createAccount(data: AccountRequest) {
        return http.post<ApiResponse<string>>(`${BASE}/add`, data);
    }
}